web/pandas/pdeps/0003-small-data-visualization-consolidation.md

```diff
@@ -0,0 +1,158 @@
+
+## Motivation and Scope
+
+Large data is not in scope for this PDEP. Large Data is classed as that which
+would not be expected to print, or be wire-transferred for example, with LaTeX,
+HTML or JSON.
+
+An important part of data analysis is data exploration and data visualization.
+`Styler` already exists as a part of Pandas, and whose original purpose was
+for conditional formatting for data cells in HTML. A rough timeline of
+activity of Styler development has been:
+  - 2015: released as conditional HTML formatting
+  - 2015-2020: expanded CSS functionality and added built in conditional
+    formatting functions.
+  - 2020: performance enhancements,
+  - 2021: expanded built in methods, expanded formatting functions,
+    performance enhancements, added LaTeX as an alternative to HTML,
+  - 2022: added concat to combine the output of multiple Stylers, documentation
+    and consistency improvements.
+
+With the feature enhancements in 2021/22 coupled with performance improvements
+it is appropriate to re-target the Styler away from an HTML conditional
+formatting only tool.
+
+The overall design philosophy for Styler is proposed as:
+
+  - i) Function as a decoupled display formatting tool.
+  - ii) Provide output methods for small data within a consistent API framework.
+  - iii) Provide enhanced formatting techniques such as hiding, color, font and border control to output methods that support it, and document these.
+  - iv) Should be the only displaying rendering method for HTML and LaTeX.
+
+## Detailed Description
+
+*Styler Philosophy - i) Decoupled display formatter:*
+
+This involves creating output without altering, or the need to pre-alter any of
+the in memory data of the DataFrame.
+This serves the purpose of maintaining high performance DataFrame indexing
+and data features without penalising display.
+
+```python
+>>> df = DataFrame({"data": range(100)},
+                   columns=date_range("2022-07-01", periods=100))
+>>> df.loc["2022-08-10": "2022-08-14"].style.format_index(lambda v: v.strftime("%a"))
+           data
+Wed   -0.245266
+Thu    0.123456
+Fri    1.123456
+Sat   -1.124578
+Sun    0.999888
+```
+
+It also allows concatenation of display without the need to concatenate DataFrames
+and avoids mixing column dtypes.
+
+```python
+>>> df = DataFrame({"norm": np.random.randn(10000),
+                "poisson": np.random.poisson(1, 10000)})
+>>> df_summary = df.agg(["mean"])
+>>> pd.options.styler.render.max_rows=5
+>>> df.style.concat(df_summary.style)
+           norm    poisson
+0     -0.111111          0
+1      0.111111          1
+2      1.234567          0
+3     -2.666777          1
+4      0.443355          2
+...         ...        ...
+mean   0.001122   0.999999
+```
+
+This design choice allows it to be separated from core functions,
+permitting a forked component in the future if ever necessary.
+
+*Styler Philosophy - ii) Output methods for small data*
```

Comment:
Most of the implementations done so far were an extension to the existing architecture. In the absense of this PDEP process I don't think I would have had much success, even if I had considered this. I think that larger, fundamental changes are harder to get through, especially if the submitter is not already in the core team.

Think this would require a more technical specification how exactly this would work across each output method. Not a big fan of such a large change at this stage though.